internal/isoeditor/rhcos.go

```diff
@@ -0,0 +1,129 @@
+package isoeditor
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/openshift/assisted-service/internal/isoutil"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// CustomIsoTempDir is a temporary directory pattern for the custom ISO
+	CustomIsoTempDir string = "customiso"
+	// BaseIsoTempDir is a temporary directory pattern for the extracted base ISO
+	BaseIsoTempDir string = "baseiso"
+	// CustomIsoObjectName is a name pattern for the ISO template
+	CustomIsoObjectName string = "rhcos-%s-minimal.iso"
+)
+
+type Editor interface {
+	CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) (string, error)
+}
+
+type rhcosEditor struct {
+	isoHandler    isoutil.Handler
+	rootfsURL     string
+	isoTmpWorkDir string
+	log           logrus.FieldLogger
+}
+
+func NewEditor(isoHandler isoutil.Handler, rootfsURL, isoTmpWorkDir string, log logrus.FieldLogger) Editor {
+	return &rhcosEditor{
+		isoHandler: isoHandler,
+		rootfsURL:  rootfsURL,
+		log:        log,
+	}
+}
+
+func CreateEditor(isoPath string, rootfsURL string, log logrus.FieldLogger) Editor {
+	isoTmpWorkDir, err := ioutil.TempDir("", BaseIsoTempDir)
+	if err != nil {
+		return nil
+	}
+	isoHandler := isoutil.NewHandler(isoPath, isoTmpWorkDir)
+	return NewEditor(isoHandler, rootfsURL, isoTmpWorkDir, log)
+}
+
+func GetMinimalIsoObject(openshiftVersion string) string {
+	return fmt.Sprintf(CustomIsoObjectName, openshiftVersion)
+}
+
+func GetRootfsURL(serviceBaseURL, openshiftVersion string) string {
+	return fmt.Sprintf("%s/api/assisted-install/v1/boot-files?file_type=rootfs.img&openshift_version=%s",
+		serviceBaseURL, openshiftVersion)
+}
+
+func (e *rhcosEditor) CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) (string, error) {
+	if err := e.isoHandler.Extract(); err != nil {
+		return "", err
+	}
+	defer os.Remove(e.isoTmpWorkDir)
+
+	if err := os.Remove(e.isoHandler.ExtractedPath("images/pxeboot/rootfs.img")); err != nil {
+		return "", err
+	}
+
+	if err := e.addRootFSURL(); err != nil {
+		e.log.WithError(err).Warnf("Can't add rootfs_url (missing cfg files)")
+	}
+
+	// create a temp dir for writing the iso there.
+	outDir, err := ioutil.TempDir("", CustomIsoTempDir)
+	if err != nil {
+		return "", err
+	}
+
+	objectName := GetMinimalIsoObject(openshiftVersion)
+	e.log.Infof("Creating minimal ISO template: %s/%s", outDir, objectName)
+	if err := e.create(ctx, objectName, outDir); err != nil {
+		return "", err
+	}
+
+	return outDir, nil
+}
+
+func (e *rhcosEditor) create(ctx context.Context, objectName, outDir string) error {
```

Comment:
Just realized `ctx` is unused here. It can be removed.

Reply:
done